features/device-config/monitor-adaptor/src/main/java/org/opennms/features/deviceconfig/monitor/adaptor/DeviceConfigMonitorAdaptor.java

```diff
@@ -0,0 +1,140 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2022 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.deviceconfig.monitor.adaptor;
+
+import com.google.common.base.Strings;
+import org.opennms.features.deviceconfig.persistence.api.ConfigType;
+import org.opennms.features.deviceconfig.persistence.api.DeviceConfig;
+import org.opennms.features.deviceconfig.persistence.api.DeviceConfigDao;
+import org.opennms.netmgt.dao.api.IpInterfaceDao;
+import org.opennms.netmgt.model.OnmsIpInterface;
+import org.opennms.netmgt.poller.MonitoredService;
+import org.opennms.netmgt.poller.PollStatus;
+import org.opennms.netmgt.poller.ServiceMonitorAdaptor;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Map;
+import java.util.Optional;
+
+public class DeviceConfigMonitorAdaptor implements ServiceMonitorAdaptor {
+
+    private static final String DEVICE_CONFIG_MONITOR = "DeviceConfig";
+
+    @Autowired
+    private DeviceConfigDao deviceConfigDao;
+
+    @Autowired
+    private IpInterfaceDao ipInterfaceDao;
+
+    @Override
+    public PollStatus handlePollResult(MonitoredService svc, Map<String, Object> parameters, PollStatus status) {
+
```

Comment:
The current implementation uses a database record to store either a success (using columns `lastUpdated/config`) or a failure (using columns `lastFailed/failureReason`). A database record does not store both of these at the same time. Maybe that should be documented.

Because of that, the `lastUpdated` column can be used to store the time of the last failure, too. The `lastFailed` column can be dropped. This would make the storage logic more apparent.